test/test.c

```diff
@@ -7170,6 +7188,53 @@ test_AuthToken(void)
     _stopServer(serverPid);
 }
 
+static const char*
```

Comment:
Not testing the reconnect use case with TokenCb. Will do so if it feels required. (The test done in https://github.com/nats-io/go-nats/pull/405/files#diff-92c931046bdf2f3c164a4a28facea12fR152)